src/containers/UserProfile.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import Input from "../components/Input/Input";
+import Button from "../components/Button/Button";
+import fetchUsers from "../utils/fetchUsers";
+import { User } from "../interfaces/interfaces.ts"
+import UserDetails from "../components/UserDetails/UserDetails.tsx";
+
+export default function UserProfile() {
+  const [user, setUser] = useState<User[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [isUserDetailsVisible, setisUserDetailsVisible] = useState<boolean>(false);
+
+  async function fetchData(inputValue: string) {
+    try {
+      const data = await fetchUsers(inputValue);
+      setUser(data);
+    } catch (error) {
+      setError(error.message);
+    }
+  }
+  if (error) {
+    return <div>Error: {error}</div>;
+  }
+  // set the input value to a variable state
+  const handleOnChange = (value) => {
+    setInputValue(value)
+  }
+
+  // call fetch data on onclick passing the input value as parameter 
+  const handleOnClick = () => {
+    fetchData(inputValue)
+    setisUserDetailsVisible(true);
```

Comment:
I believe you don't need this `setIsUserDetailsVisible` state. Imagine the case where your `fetchData` function fails, then you need to show an error instead of the user details. So you need to render conditionally based onto the `error` state or the `data` state.